frame/composable-traits/src/dex.rs

```diff
@@ -22,6 +22,35 @@ pub trait CurveAmm {
 	/// Current number of pools (also ID for the next created pool)
 	fn pool_count() -> Self::PoolId;
 
+	/// Get pool token index for given asset_id.
+	fn get_token_index(
```

Comment:
i am concerned that we have impl for pair and api for multi pool.

did we talked about to have dexes only for pairs? cc @hussein-aitlahcen

If yes, than we can have all api for pairs. but later can have impl for multi (router, balancer, curve multi) are multi dexes.

Reply:
An option would be to split up the trait into multiple ones, as is done with Mutate, Transfer etc.